omnigibson/simulator.py

```diff
@@ -238,12 +258,28 @@ def __init__(
             stage_units_in_meters=1.0,
             viewer_width=gm.DEFAULT_VIEWER_WIDTH,
             viewer_height=gm.DEFAULT_VIEWER_HEIGHT,
+            floor_plane_visible=True,
+            floor_plane_color=(1.0, 1.0, 1.0),
+            use_skybox=True,
             device=None,
         ):
+            # Here we assign self as the Simulator instance and as og.sim, because certain functions
+            # called downstream during the initialization of this object will try to access og.sim.
+            # This makes that possible (and also introduces possible issues around circular dependencies)
```

Comment:
Here lets also check that there is not a stage open.

Reply:
Note that we dont explicitly craete one here - super.init does. We just need to check that one doesnt already exis.t

Reply:
It seems like `SimulationApp` creates a stage before this:
```
        self.open_usd = self.config.get("open_usd")
        if self.open_usd is not None:
            print("Opening usd file at ", self.open_usd, " ...", end="")
            if open_stage(self.open_usd) is False:
                print("Could not open", self.open_usd, "creating a new empty stage")
                create_new_stage()
            else:
                print("Done.")
        else:
            create_new_stage()
```
Do you think we should clear that stage, assert there's no stage, and than open a stage in Simulator init?

Reply:
Is it possible to "close" a stage and leave the app stageless? If so, let's do that between creating SimulationApp and Simulator, such that the Simulator init gets a stageless state. Otherwise let's just assert that nothing (unexpected) is in the stage at Simulator init time.